specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-04-01-preview/workspaceRP.json

```diff
@@ -0,0 +1,6198 @@
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ConnectionName"
+          },
+          {
+            "$ref": "#/parameters/DeploymentNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the request was well-formed and received properly.",
+            "schema": {
+              "$ref": "#/definitions/EndpointDeploymentResourcePropertiesBasicResource"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Azure OpenAI Connection Deployment": {
+            "$ref": "./examples/WorkspaceConnection/getDeployment.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "V2WorkspaceConnectionResource"
+        ],
+        "summary": "Create or update Azure OpenAI connection deployment resource with the specified parameters",
+        "operationId": "Connection_CreateOrUpdateDeployment",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ConnectionName"
+          },
+          {
+            "$ref": "#/parameters/DeploymentNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "deployment object",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/EndpointDeploymentResourcePropertiesBasicResource"
+            }
+          }
+        ],
+        "responses": {
```

Comment:
Synchronous and long-running PUT operations must have responses with 200, 201 and default return codes. They must not have any other response codes.
https://github.com/Azure/azure-openapi-validator/blob/main/docs/put-response-codes.md

Reply:
we would like to request a exception for this API, we will have all our new API following this requirement, but this API is used to migrate from this below API, so we hope we can keep the return status code the same to avoid any breaking on the caller.
https://github.com/Azure/azure-rest-api-specs/blob/ead20be2d8d680dfee3605e6f18148e65cbb79e4/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json#L1267

Reply:
Please use the opportunity to make the implementation consistent with the guidelines and the rest of your resource types.

Reply:
This requires major refactoring on both client and service side, the current 202 implementation has already been released and in use, so hope we can get an exception on this one to unblock this API version.

Reply:
https://github.com/Azure/azure-rest-api-specs/pull/28495

this is the PR to change to 201 status code